spec/utility/bulk_queue_spec.rb

```diff
@@ -0,0 +1,159 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+require 'utility/bulk_queue'
+
+describe Utility::BulkQueue do
+  let(:subject) { described_class.new(count_threshold, size_threshold) }
+  let(:count_threshold) { 50 }
+  let(:size_threshold) { 999999999 } # super high default
+
+  context '.add' do
```

Comment:
[nitpick] It's better to use #describe when you start testing a method.

refs: https://www.betterspecs.org/